### PR TITLE
feat: Added proper wait, logging and snapshots

### LINKED ISSUE #2: Cause `.pipe` to retry automatically when the function returns an empty element list
```ts
const getFoo = $el => $el.find('.foo') // <div class="foo"> appears after some time

cy.get('body')  // waits for a `body` tag to exist
  .pipe(getFoo) // won't retry unless followed by a `.should('be.visible')`
```

### PR DESCRIPTION
**Pull request by NicholasBoll**
Add proper logging and snapshots around piped helper functions:

screen shot 2018-09-27 at 4 15 35 pm

Fixes #2 and #3

### CODE REVIEW

File: cypress.json [hunk 1]
```diff
@@ -1 +1,3 @@
-{}
+{
+  "baseUrl": "http://localhost:3333"
```
Comment (pnaza) [inline cypress.json:2]: is this needed? if so, should  it be `localhost:8443` ?

File: package.json [hunk 2]
```diff
@@ -28,6 +29,11 @@
   },
   "homepage": "https://github.com/NicholasBoll/cypress-pipe#readme",
   "devDependencies": {
-    "cypress": "^2.1.0"
+    "cypress": "^3.1.0",
+    "http-server": "^0.11.1"
+  },
+  "scripts": {
+    "start": "http-server . -p 3333",
```
Comment (pnaza) [inline package.json]: what is this for?

**Review by pnaza — APPROVED** [review]

File: cypress.json (hunk 1 above)
Comment (NicholasBoll) [reply cypress.json:2]: No, this `3333` is the webserver for this project

File: package.json (hunk 2 above)
Comment (NicholasBoll) [reply package.json]: That starts a web server for the tests on this project

**Review by andy-lindberg — APPROVED** [review]

**Comment by NicholasBoll** [comment]
@andy-lindberg @pnaza I added tests to ensure the DOM snapshots are being made. I also updated the README with more info on these changes.

**Review by pnaza — APPROVED** [review]

**Pull request merged**